exporters/prometheus/exporter.go

```diff
@@ -181,6 +183,16 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		ch <- c.targetInfo
 	}
 
+	var resourceAttrs []attribute.KeyValue
```

Comment:
can we cache this so we don't have to recompute them each time?

Reply:
are you thinking this could be done at the same time as the target info is initialized above?

Reply:
i've updated this to cache the results using the resource as the key, PTAL